apps/aeutils/src/aeu_split_check.erl

```diff
@@ -0,0 +1,54 @@
+-module(aeu_split_check).
+
+-export([
+         run_for_mainnet/1,
+         run_for_testnet/1,
+         balance_mainnet/1,
+         balance_testnet/1,
+         balance/2,
+         rewards/1
+        ]).
+
+-include_lib("aecontract/include/hard_forks.hrl").
+
+-define(TESTNET_ACC, <<"ak_2A3PZPfMC2X7ZVy4qGXz2xh2Lbh79Q4UvZ5fdH7QVFocEgcKzU">>).
+-define(MAINNET_ACC, <<"ak_2KAcA2Pp1nrR8Wkt3FtCkReGzAi8vJ9Snxa4PcmrthVx8AhPe8">>).
+
+run_for_mainnet(H) ->
+    balance_mainnet(H) =:= rewards(H).
+
+run_for_testnet(H) ->
+    balance_testnet(H) =:= rewards(H).
+
+balance_mainnet(H) ->
+    balance(H, ?MAINNET_ACC).
+
+balance_testnet(H) ->
+    balance(H, ?TESTNET_ACC).
+
+balance(H, Account) ->
+    {ok, HashNext} = aec_chain:get_key_header_by_height(H+1),
+    Hash = aec_headers:prev_key_hash(HashNext),
+    {ok, Trees} = aec_chain:get_block_state(Hash),
+    AccountTrees = aec_trees:accounts(Trees),
+    {_, Kbin} = aeser_api_encoder:decode(Account),
+    RewardAccount = aec_accounts:balance(aec_accounts_trees:get(Kbin, AccountTrees)),
+    RewardAccount.
+
+rewards(H) ->
```

Comment:
```suggestion
%% Return cumulative block reward at specified height.
%%
%% Assumes no fork switch during the computation.
rewards(H) ->
```

Reply:
Fork switch is atomic, so while there is indeed a race condition between computation and presentation it is always valid in some context. The context is a chain that is pulled via gen n headers api.